src/quo2/components/markdown/list/view.cljs

```diff
@@ -0,0 +1,33 @@
+(ns quo2.components.markdown.list.view
+  (:require [react-native.core :as rn]
+            [quo2.components.markdown.text :as text]
+            [quo2.components.counter.step.view :as step]
+            [quo2.components.markdown.list.style :as style]
+            [quo2.components.icon :as icon]
+            [quo2.foundations.colors :as colors]
+            [quo2.theme :as theme]))
+
+(defn- themed-view
+  [{:keys [theme title description index step-props]}]
```

Comment:
I want to suggest that instead of passing `step-props` and rely on step's component API, this component exposes its own API.

for example, the properties that I think we would be adding are:
- `:type` -> bullet | step
- `:step-type` -> `:neutral` | `:complete`
- `:step-number` (or the name you pick)
- `:customization-color`

so:
```clojure
;; from:
[quo/markdown-list {:title       "My title"
                    :description "My description"}]  ;; We don't know the type here

;; to:
[quo/markdown-list {:title       "My title"
                    :description "My description"
                    :type        :bullet}] ;; type bullet added, as in figma
```

```clojure
;; from:
[quo/markdown-list {:title       "My title"
                    :description "My description"
                    :step-number 15
                    :step-props  {:type :neutral}}] ;; We get rid of this nested map

;; to:
[quo/markdown-list {:title       "My title"
                    :description "My description"
                    :step-number 15
                    :type        :step
                    :step-type   :neutral}]
```

and

```clojure
;; from:
[quo/markdown-list {:title       "My title"
                    :description "My description"
                    :step-number 15
                    :step-props  {:type                :complete
                                  :customization-color :blue}}]
                                   ;; IMO the custom color is applied to the whole
                                   ;; component, so looks better if we extract it

;; to:
[quo/markdown-list {:title               "My title"
                    :description         "My description"
                    :step-number         15
                    :type                :step
                    :step-type           :complete
                    :customization-color :blue}]
```

What do you think @J-Son89 ?
I'm just trying to treat this component as a whole itself instead of "composing" its parts through nested maps

Reply:
yep, that all sounds reasonable and I agree it is clearer. It's a simple enough component anyway so it benefits from the flatter api 👍